cmd/podman/utils.go

```diff
@@ -215,3 +216,50 @@ func getPodsFromContext(c *cli.Context, r *libpod.Runtime) ([]*libpod.Pod, error
 	}
 	return pods, lastError
 }
+
+type pFunc func() error
+
+type workerInput struct {
+	containerID  string
+	parallelFunc pFunc
+}
+
+// worker is a "threaded" worker that takes jobs from the channel "queue"
+func worker(wg *sync.WaitGroup, jobs <-chan workerInput, results map[string]error) {
+	for j := range jobs {
+		err := j.parallelFunc()
+		results[j.containerID] = err
+		wg.Done()
+	}
+}
+
+// parallelExecuteWorkerPool takes container jobs and performs them in parallel.  The worker
+// int is determines how many workers/threads should be premade.
+func parallelExecuteWorkerPool(workers int, functions []workerInput) map[string]error {
+	var (
+		wg sync.WaitGroup
+	)
+	results := make(map[string]error)
+	paraJobs := make(chan workerInput, len(functions))
+
+	// If we have more workers than functions, match up the number of workers and functions
+	if workers > len(functions) {
+		workers = len(functions)
+	}
+
```

Comment:
If you end up with more functions than workers, are they handled?

Reply:
ofc, thats the parajobs queue

Reply:
DOH!  TY!  Somehow missed the len(functions) in the paraJobs def.  Back to the tea pot for me.

Reply:
The range loop in the worker keeps accepting jobs until the channel empties